### DISCUSSION TITLE
Virtual threads support

### DISCUSSION THREAD

**Discussion by oclay1st** [jobrunr/jobrunr#838] [Ideas]
Are you open to support virtual threads as an open source feature?. Are you open to receive help? Is there any open related discussion (I did some research, zero result so far)?. Thanks

**Comment by rdehuyss** [comment]
Well, we already researched Virtual Threads back in August, 2020 - see https://www.jobrunr.io/en/blog/2020-08-17-jobrunr-loom-virtual-threads/.

We're doing multiple performance initiatives for the moment:
- Currently, we're preparing everything for an external review of JobRunr by a Java Champion and Java Performance specialist. There we will look together to see whether changing the default threading in Java 21 makes sense. If Virtual Threads make sense (which I currently don't know yet), we will also add it to JobRunr OSS.
- For JobRunr Pro, we are adding machine learning capabilities where JobRunr will automatically adjust the number of threads automatically to optimize throughput.

**Reply by oclay1st** [reply]
Looking forward to hear more about this in the near future. I hope you can make it optional and also embrace the new integrations with frameworks like Spring Boot 3.2, Quarkus, etc. Thanks

**Reply by ilopmar** [reply]
Thanks for this! I'm upgrading my Spring application to SB 3.2.0-RC2 and I wanted to give it a try to the Virtual Threads support (if it was there).
Looking forward to try it if it's finally included!

**Reply by drakiula** [reply]
It is actually quite trivial to follow the provided sample from that blog post and use the open source version with the `VirtualThreadJobRunrExecutor`, I am using it in a project with latest JobRunr open source and spring boot 3.2.2.

All needed details can be found here, 
https://github.com/jobrunr/jobrunr-loom

**Reply by rdehuyss** [reply]
JobRunr 7, due out in month are so will natively support Virtual Threads.

**Reply by drakiula** [reply]
@rdehuyss will you also publish a new version for the spring boot starter?

**Reply by rdehuyss** [reply]
Yes, will be part of the new release.
